--- Reusable-Cluster-Components-glue--glue-1.0.12/configure.ac.orig	2015-12-15 22:12:07.252626084 +0100
+++ Reusable-Cluster-Components-glue--glue-1.0.12/configure.ac	2015-12-15 22:14:16.329287335 +0100
@@ -1223,10 +1223,8 @@
 fi
 
 if test "$GCC" != yes; then
-        CFLAGS="$CFLAGS -g"
 	enable_fatal_warnings=no
 else
-        CFLAGS="$CFLAGS -ggdb"
 
 	# We had to eliminate -Wnested-externs because of libtool changes
         EXTRA_FLAGS="-fgnu89-inline
