--- Reusable-Cluster-Components-glue--glue-1.0.11/configure.ac.orig	2012-10-15 15:58:55.000000000 +0200
+++ Reusable-Cluster-Components-glue--glue-1.0.11/configure.ac	2013-09-06 16:26:42.523189874 +0200
@@ -1199,10 +1199,8 @@
 fi
 
 if test "$GCC" != yes; then
-        CFLAGS="$CFLAGS -g"
 	enable_fatal_warnings=no
 else
-        CFLAGS="$CFLAGS -ggdb3 -O0"
 
 	# We had to eliminate -Wnested-externs because of libtool changes
         EXTRA_FLAGS="-fgnu89-inline
