--- heartbeat-2.1.2/lib/plugins/stonith/Makefile.am~	2007-08-02 02:06:55.000000000 +0000
+++ heartbeat-2.1.2/lib/plugins/stonith/Makefile.am	2007-09-06 13:02:10.000000000 +0000
@@ -88,7 +88,7 @@
 bladehpi_LIB =
 endif
 
-noinst_PROGRAMS = $(ipmilan_TEST)
+noinst_PROGRAMS = 
 ipmilantest_SOURCES = ipmilan_test.c 
 ipmilantest_LDADD = $(top_builddir)/lib/pils/libpils.la \
 		    $(top_builddir)/lib/stonith/libstonith.la \
